src/site/content/en/blog/predictive-prefetching/index.md

```diff
@@ -0,0 +1,109 @@
+- Some resources the user needs may never be prefetched.
+
+Predictive prefetching solves these problems by using a report of users' navigational patterns to determine what assets to prefetch.
+
+![Prefetching example](guess-1.png)
+
+## Predictive prefetching with Guess.js
+
+[Guess.js](https://github.com/guess-js) is a JavaScript library that provides predictive prefetching functionality. Guess.js consumes a report from Google Analytics or another analytics provider to build a predictive model that can be used to smartly prefetch only what the user is likely to need.
+
+Guess.js has integration with [Angular](https://angular.io), [Next.js](https://nextjs.org/), [Nuxt.js](https://nuxtjs.org/), and [Gatsby](https://www.gatsbyjs.org/). To use it in your application, you should add a single line of webpack configuration specifying a [Google Analytics view ID](https://stackoverflow.com/questions/36898103/what-is-a-viewid-in-google-analytics):
+
+```js/0,5
+const { GuessPlugin } = require('guess-webpack');
+
+// ...
+plugins: [
+   // ...
+   new GuessPlugin({ GA: 'XXXXXX' })
+]
+// ...
+```
+
+If you're not using Google Analytics, you can specify a `reportProvider` and download data from your favorite service.
+
+### Integration with frameworks
+
+To learn more about how to integrate Guess.js with your favorite framework check out these resources:
+
+1. [Using Guess.js with Angular](https://guess-js.github.io/docs/angular)
+1. [Using Guess.js with Next.js](https://guess-js.github.io/docs/next)
+1. [Using Guess.js with Nuxt.js](https://guess-js.github.io/docs/nuxt)
+
+For a quick walkthrough on the integration with Angular, check out this video:
+
+{% YouTube '5FRxQiGqqmM' %}
+
+### How does Guess.js work?
+
+Here's how Guess.js implements predictive prefetching:
+
+1. It first extracts data for the user navigational patterns from your favorite analytics provider.
+1. It then maps the URLs from the report to the JavaScript chunks produced by webpack.
+1. Based on the extracted data, it creates a simple predictive model of which pages a user is likely to navigate to from any given page.
+1. It invokes the model for each JavaScript chunk, predicting the other chunks that are likely to be needed next.
+1. It adds prefetching instructions to each chunk.
+
+In the end, each chunk will contain prefetching instructions similar to:
+
+```js
+__GUESS__.p(
+  ['a.js', 0.2],
+  ['b.js', 0.8]
+)
+```
+
+The meaning behind this code that Guess.js generated is: consider prefetching chunk `a.js` with probability `0.2` and chunk `b.js` with probability `0.8`.
+
+Once the browser executes this code Guess.js will first check the user's connection speed. If it's sufficient enough Guess.js will insert two `<link rel="prefetch">` tags in the header of the page, one for each chunk. If the user has a poor network connection, Guess.js will only prefetch the chunk `b.js` since it has `0.8` probability to be needed next.
```

Comment:
```suggestion
Once the browser executes the code, Guess.js will check the user's connection speed. If it's sufficient, Guess.js will insert two `<link rel="prefetch">` tags in the header of the page, one for each chunk. If the user is on a high-speed network, Guess.js will prefetch both chunks. If the user has a poor network connection, Guess.js will only prefetch chunk `b.js` since it has a high probability of being needed.
```
Fleshed this out to try to make the logic a bit more explicit. See what you think.